src/coreclr/jit/valuenum.cpp

```diff
@@ -2136,7 +2136,7 @@ ValueNum ValueNumStore::VNForFunc(
     assert(arg0VN == VNNormalValue(arg0VN));
     assert(arg1VN == VNNormalValue(arg1VN));
     assert(arg2VN == VNNormalValue(arg2VN));
-    assert(arg3VN == VNNormalValue(arg3VN));
+    assert((func == VNF_MapStore) || (arg3VN == VNNormalValue(arg3VN)));
```

Comment:
Comment above: "// Note:   Currently the only four operand func is the VNF_PtrToArrElem operation" is no longer true

Reply:
Thanks, fixed in #56184.